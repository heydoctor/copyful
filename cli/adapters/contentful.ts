import 'dotenv/config';
import set from 'lodash.set';
import { createClient } from 'contentful';

export const client = createClient({
  space: process.env.COPYFUL_CONTENTFUL_SPACE,
  accessToken: process.env.COPYFUL_CONTENTFUL_ACCESS_TOKEN,
});

export default async function clone() {
  const { items } = await client.getEntries<{ category: string; name: string; content: string }>({
    content_type: 'copyBlock',
    limit: 1000,
  });

  return items.reduce((copy, { fields }) => {
    return set(copy, `en-us.${fields.category}.${fields.name}`, fields.content);
  }, {});
}
