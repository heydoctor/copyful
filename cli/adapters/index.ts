import contentful from './contentful';

export { contentful };
