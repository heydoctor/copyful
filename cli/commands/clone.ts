import { writeFileSync } from 'fs';
import * as adapters from '../adapters';

export default async function clone({ adapter, path }: { adapter: 'contentful'; path: string }) {
  const data = await adapters[adapter]();
  writeFileSync(path, `export const copy = ${JSON.stringify(data, null, 2)} as const;`);
}
