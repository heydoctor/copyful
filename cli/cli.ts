#! /usr/bin/env node

import * as yargs from 'yargs';
import clone from './commands/clone';

yargs
  // @ts-ignore
  .scriptName('copyful')
  .usage('$0 <cmd> [args]')
  .command(
    'clone',
    'Pull down copy from remote source and write to file system',
    yargs => {
      yargs.positional('adapter', {
        type: 'string',
        default: 'contentful',
        describe: 'Remote source',
      });
      yargs.positional('path', {
        type: 'string',
        describe: 'Destination file',
      });
    },
    clone
  )
  .help().argv;
