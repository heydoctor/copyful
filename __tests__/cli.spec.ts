import 'regenerator-runtime/runtime';
import { readFileSync } from 'fs';
import clone from '../cli/commands/clone';
import * as adapters from '../cli/adapters';

jest.mock('contentful', () => ({
  createClient() {
    return {
      getEntries() {
        return {
          sys: { type: 'Array' },
          total: 1,
          skip: 0,
          limit: 100,
          items: [
            {
              sys: { type: 'Entry' },
              fields: { category: 'test', name: 'hola', content: 'mundo' },
            },
          ],
        };
      },
    };
  },
}));

describe('cli', () => {
  describe('commands', () => {
    describe('clone()', () => {
      const testClonePath = '/tmp/copyful-clone-test.js';

      beforeEach(() => {
        jest
          .spyOn(adapters, 'contentful')
          .mockImplementation(() => Promise.resolve({ test: 'data' }));
      });

      afterEach(() => {
        jest.restoreAllMocks();
      });

      it('writes data returned from the adapter to the file system', async () => {
        await clone({ adapter: 'contentful', path: testClonePath });

        expect(readFileSync(testClonePath, { encoding: 'utf8' })).toMatchSnapshot();
      });
    });
  });

  describe('adapters', () => {
    describe('contentful', () => {
      it('transforms data into expected structure', () => {
        return expect(adapters.contentful()).resolves.toMatchSnapshot();
      });
    });
  });
});
