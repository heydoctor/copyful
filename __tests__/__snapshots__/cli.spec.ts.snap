// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli adapters contentful transforms data into expected structure 1`] = `
Object {
  "en-us": Object {
    "test": Object {
      "hola": "mundo",
    },
  },
}
`;

exports[`cli commands clone() writes data returned from the adapter to the file system 1`] = `
"export const copy = {
  \\"test\\": \\"data\\"
} as const;"
`;
